### PR TITLE
Expose sygus assumptions and constraints to the api

### PR DESCRIPTION
**Pull request by mudathirmahgoub**

### CODE REVIEW

File: src/api/cpp/cvc5.cpp [hunk 1]
```diff
@@ -7822,6 +7832,16 @@ void Solver::addSygusAssume(const Term& term) const
   CVC5_API_TRY_CATCH_END;
 }
 
+std::vector<Term> Solver::getSygusAssumptions() const
+{
+  CVC5_API_TRY_CATCH_BEGIN;
+  //////// all checks before this line
+  std::vector<internal::Node> constraints = d_slv->getSygusAssumptions();
+  return Term::nodeVectorToTerms(d_nm, constraints);
```
Comment (abdoo8080) [inline src/api/cpp/cvc5.cpp:7840]:
```suggestion
  std::vector<internal::Node> assumptions = d_slv->getSygusAssumptions();
  return Term::nodeVectorToTerms(d_nm, assumptions);
```

**Review by abdoo8080 — COMMENTED** [review]
Are we exposing this data because of oracles?

**Comment by mudathirmahgoub** [comment]
> Are we exposing this data because of oracles?

I needed it to compare a problem given to z3 spacer solver versus the problem given to cvc5 sygus solver.
The api already has support for `getAssertions`, so it makes sense to have the same for sygus.

**Review by ajreynol — APPROVED** [review]
LGTM, thanks for adding this.

**Pull request merged**